src/util.hpp

```diff
@@ -52,59 +52,49 @@ mapbox::geometry::point<std::int64_t> create_query_point(double lng,
-    std::uint64_t origin_tile_y = static_cast<std::uint64_t>(floor(zl_y / extent));
-    // std::clog << "origin tile x: " << origin_tile_x << ", origin tile y: " << origin_tile_y << std::endl;
-    // std::clog << "active tile x: " << active_tile_x << ", active tile y: " << active_tile_y << std::endl;
-
-    std::uint64_t origin_x = std::uint32_t(fmod(floor(zl_x), extent));
-    std::uint64_t origin_y = std::uint32_t(fmod(floor(zl_y), extent));
-    // std::clog << "origin tile coordinate x: " << origin_x << ", origin tile coordinate y: " << origin_y << std::endl;
-
-    auto diff_tile_x = active_tile_x - origin_tile_x;
-    auto diff_tile_y = active_tile_y - origin_tile_y;
-    // std::clog << "tile difference x: " << diff_tile_x << ", y: " << diff_tile_y << std::endl;
-    auto query_x = -(diff_tile_x * extent) + origin_x;
-    auto query_y = -(diff_tile_y * extent) + origin_y;
-    // std::clog << "query point x: " << query_x << ", y: " << query_y << std::endl;
-    mapbox::geometry::point<std::int64_t> query_point{static_cast<std::int64_t>(query_x), static_cast<std::int64_t>(query_y)};
+    double zl_y = ((extent * z2) / 2) * (1.0 - (std::log(std::tan(lat_radian) + 1.0 / std::cos(lat_radian)) / M_PI));
+    std::int64_t origin_tile_x = static_cast<std::int64_t>(std::floor(zl_x / extent));
+    std::int64_t origin_tile_y = static_cast<std::int64_t>(std::floor(zl_y / extent));
+    std::int64_t origin_x = std::int64_t(std::fmod(std::floor(zl_x), extent));
+    std::int64_t origin_y = std::int64_t(std::fmod(std::floor(zl_y), extent));
+    std::int64_t diff_tile_x = active_tile_x - origin_tile_x;
+    std::int64_t diff_tile_y = active_tile_y - origin_tile_y;
+    std::int64_t query_x = -(diff_tile_x * extent) + origin_x;
+    std::int64_t query_y = -(diff_tile_y * extent) + origin_y;
+    mapbox::geometry::point<std::int64_t> query_point{query_x, query_y};
     return query_point;
 }
 
 /*
- input: z/x/y tile address, x,y within tile
- returns: std::pair<x, y> in wgs84
+  Create a geometry.hpp point from vector tile coordinates
 */
 template <typename CoordinateType>
-std::pair<double, double> convert_vt_to_ll(std::uint32_t extent,
-                                           std::uint32_t z,
-                                           std::uint32_t x,
-                                           std::uint32_t y,
-                                           CoordinateType px,
-                                           CoordinateType py) {
-    double size = extent * std::pow(2.0, z);
-    double x0 = static_cast<double>(extent) * x;
+mapbox::geometry::point<double> convert_vt_to_ll(std::uint32_t extent,
+                                                 std::uint32_t z,
+                                                 std::uint32_t x,
+                                                 std::uint32_t y,
+                                                 CoordinateType px,
```

Comment:
Convert this to take an input of `mapbox::geometry::point<T>` rather then a `x,y`

Reply:
This means I'll have to convert the output of closeset_point `x` and `y` into a mapbox::geometry::point. How about passing in the `closest_point_info` structure instead?

Reply:
Added a `mapbox::geometry::algorithms::closet_point_info` type here